prow/plugins/project/project.go

```diff
@@ -0,0 +1,270 @@
+	m := make(map[string]int)
+	for _, project := range projects {
+		m[project.Name] = project.ID
+	}
+	return m
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfigMap map[string]plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	matches := projectRegex.FindStringSubmatch(e.Body)
+	if len(matches) != 2 && len(matches) != 3 {
+		return nil
+	}
+
+	proposedProject := matches[1]
+	proposedColumn := -1
+	if len(matches) == 3 && proposedProject != clearKeyword {
+		columnInt, err := strconv.Atoi(matches[2])
+		if err == nil {
+			proposedColumn = columnInt
+		}
+	}
+
+	// See if the current user is part of the team in order to set the project/column
+	projectConfig, exists := projectConfigMap[fmt.Sprintf("%s/%s", org, repo)]
+	if !exists {
+		// fallback default
+		projectConfig = projectConfigMap[""]
+	}
+	projectMaintainers, err := gc.ListTeamMembers(projectConfig.ProjectMaintainersID, github.RoleAll)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
+		login := github.NormLogin(e.User.Login)
+		if github.NormLogin(person.Login) == login {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// not in the project maintainers team
+		msg := fmt.Sprintf(mustBeSigLead, org, project.ProjectMaintainersTeam, org, project.ProjectMaintainersTeam)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	var projects []Project
+
+	repos, err := gc.GetRepos(org, false)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all repos in org %s", org)
+		return err
+	}
+	// Get all projects for all repos
+	for _, repo := range repos {
+		repoProjects, err := gc.GetRepoProjects(org, repo.Name)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all projects under repo %s", repo.Name)
+			return err
+		}
+		projects = append(projects, repoProjects)
+	}
+
+	// Get all projects for this org
+	orgProjects, err := gc.GetOrgProjects(org)
+	if err != nil {
+		log.WithError(err).Errorf("Error getting projects for org %s", org)
+		return err
+	}
+	projects = append(projects, orgProjects)
+
+	// Get all columns for each project
+	projectColumnMap map[string][]github.ProjectColumn
+	for _, project := range projects {
+		projectColumns, err := gs.GetProjectColumns(project.ID)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all columns under project %s", project.Name)
+			return err
+		}
+		projectColumnMap[project.Name] = projectColumns
+	}
+
+	projectMap := buildProjectMap(projects)
+	projectID, ok := projectMap[proposedProject]
+
+	if proposedProject == clearKeyword {
+		projectToClear := matches[2]
+		projectToClearID, ok := projectMap[projectToClear]
+		if ok {
+			existingProjectCard, err := gc.GetProjectCard(projectToClearID, e.Number)
```

Comment:
Why do we need his call if we already have the ID?

Reply:
We have the project ID but we need to project card in order to move the project card